### PR TITLE
Allow signing subkeys in rnp_generate_key_ex().

### LINKED ISSUE #1527: rnp_generate_key_ex() doesn't allow to generate signing subkey.
## Description

Currently it defaults to encrypt-only subkey, however it may check supplied algorithm and if it is sign-only then change subkey's usage to sign instead of failing.

### PR DESCRIPTION
**Pull request by antonsviridenko**
Closes #1527

### CODE REVIEW

File: src/lib/rnp.cpp [hunk 1]
```diff
@@ -5092,8 +5092,17 @@ try {
...
-        goto done;
+    if (pgp_pk_alg_capabilities(subop->crypto.key_alg) & PGP_KF_ENCRYPT) {
+        if ((ret = rnp_op_generate_add_usage(subop, "encrypt"))) {
+            goto done;
+        }
+    } else {
+        if ((ret = rnp_op_generate_add_usage(subop, "sign"))) {
+            goto done;
+        }
+        if ((ret = rnp_op_generate_add_usage(subop, "certify"))) {
```
Comment (ni4) [inline src/lib/rnp.cpp:5103]: @antonsviridenko 
Not sure whether we should add certification flag by default, as these are different operations. Is it something which GnuPG does for signing subkey?

Comment (antonsviridenko) [reply src/lib/rnp.cpp:5103]: Yes, right, "certify" is set only for primary keys, signing subkeys have signing flag only in GnuPG.

**Review by ni4 — APPROVED** [review]
LGTM, thanks!

**Review by ronaldtse — APPROVED** [review]

**Comment by ronaldtse** [comment]
Thank you @antonsviridenko !

**Pull request merged**